### ISSUE TITLE
use native timestamp data to bootstrap TextGrid

### ISSUE THREAD

**Issue by jonorthwash** [SwatPhonLab/UltraTrace#90]
When a DICOM + audio file pair is loaded, a TextGrid should be automatically generated (kind of working already, see #72).  This TextGrid should be created with intervals based on the timestamps in the Philips data, and should be saved immediately to the directory that the existing files are in.

The method already in place should be a fallback if there is not native frametime data or if there's a problem reading it.

We may still have the problem of not knowing the relation between the frametimes and the start/end of the audio recording.  This issue could probably use some discussion.

**Comment by mr-martian** [comment]
For DICOM, timestamps generated by us are 16.44ms earlier than timestamps from the matlab script (at least, for the example data). (We currently set the first timestamp to 0.)

**Comment by jonorthwash** [comment]
This is because in the example data the alignment of frames was done from the end of the recording.  And given that, I don't know if the last frame was decided as being right at the end or what.

**Comment by keggsmurph21** [comment]
> When a DICOM + audio file pair is loaded, a TextGrid should be automatically generated (kind of working already, see #72). This TextGrid should be created with intervals based on the timestamps in the **Philips data**, and should be saved immediately to the directory that the existing files are in.
> 
> The method already in place should be a fallback if there is not native frametime data or if there's a problem reading it.
> 
> We may still have the problem of not knowing the relation between the frametimes and the start/end of the audio recording. This issue could probably use some discussion.

@jonorthwash what do you mean by "Philips data" ?

> For DICOM, timestamps generated by us are 16.44ms earlier than timestamps from the matlab script (at least, for the example data). (We currently set the first timestamp to 0.)

@mr-martian which matlab script?

**Comment by mr-martian** [comment]
@keggsmurph21 The Phillips data is the DICOM files (in contrast to ULT). My understanding is that most of the timestamp data we currently have comes from a proprietary Matlab script that we got from Phillips.

**Comment by kmurphy4** [comment]
Where does that script live? On your computer?

**Comment by mr-martian** [comment]
I was told that it only worked on Windows and I haven't ever touched it.

**Comment by jonorthwash** [comment]
> > @jonorthwash what do you mean by "Philips data" ?
> 
>  The Phillips data is the DICOM files

The Philips data is the raw data we read from DICOMs.  We can also read the DICOM pixel data natively or cache it as PNGs.  The pixel data is not Philips data; the "raw" ultrasound data is.

> > @mr-martian which matlab script?
> 
> most of the timestamp data we currently have comes from a proprietary Matlab script that we got from Phillips.

Well, we used a library from Philips that only runs on (old versions of?) Matlab for Windows to dump timestamps for each frame, and we aligned those with the end of files in TextGrids (the frames tiers).

However, @mr-martian figured out how to read the timestamp data directly, and I guess has worked on this issue regarding injecting it into TextGrid files too.  But he wasn't aligning it with the end, which is probably why there's a difference in alignment.
